src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,180 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import core.basesyntax.service.InvalidDataException;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceTest {
+    private static final int ADULT_AGE = 18;
+    private static final int ANOTHER_ADULT_AGE = 25;
+    private static final int NEGATIVE_AGE = -10;
+    private static final int TEEN_AGE = 17;
+    private static final int ZERO_AGE = 0;
```

Comment:
These constants are hard-coded age values, which might change over time. Consider removing them and using the `MIN_AGE` constant from the `RegistrationServiceImpl` class instead, if such a constant is defined there to avoid tying the tests to potentially changeable values.